src/eckit/mpi/Group.cc

```diff
@@ -32,7 +32,7 @@ class NullGroupContent : public GroupContent {
 
     virtual NullGroupContent* union_group(const GroupContent&) const { return new NullGroupContent(); }
 
-    virtual size_t size() const { return -1; }
+    virtual size_t size() const { return 0; }
```

Comment:
Whoa good catch -- I'm surprised a big fish like this stood for so long!

Reply:
It's in NullGroupContent which has other mechanisms to check that the group is not valid. Probably one never called this function.